services/MerchStoreService.ts

```diff
@@ -431,34 +441,236 @@ export default class MerchStoreService {
     if (user.credits < totalCost) throw new UserError('You don\'t have enough credits for this order');
   }
 
-  public async updateOrderItems(fulfillmentUpdates: OrderItemFulfillmentUpdate[]): Promise<void> {
-    const updates = new Map<string, OrderItemFulfillmentUpdate>();
-    for (let i = 0; i < fulfillmentUpdates.length; i += 1) {
-      const oi = fulfillmentUpdates[i];
-      oi.fulfilled = Boolean(oi.fulfilled);
-      updates.set(oi.uuid, oi);
-    }
-    await this.transactions.readWrite(async (txn) => {
-      const orderItemRepository = Repositories.merchOrderItem(txn);
-      const orderItems = await orderItemRepository.batchFindByUuid(Array.from(fulfillmentUpdates.map((oi) => oi.uuid)));
-      if (orderItems.size !== fulfillmentUpdates.length) {
-        throw new NotFoundError('Missing some order items');
+  private static isLessThanTwoDaysBeforePickupEvent(pickupEvent: OrderPickupEventModel): boolean {
+    return new Date() > moment(pickupEvent.start).subtract(2, 'days').toDate();
+  }
+
+  public async editMerchOrderPickup(orderUuid: Uuid, pickupEventUuid: Uuid, user: UserModel): Promise<OrderModel> {
+    return this.transactions.readWrite(async (txn) => {
+      const orderRepository = Repositories.merchOrder(txn);
+      const order = await orderRepository.findByUuid(orderUuid);
+      if (!order) throw new NotFoundError('Order not found');
+      if (order.user.uuid !== user.uuid) throw new ForbiddenError('Cannot edit the order of a different user');
+      if (MerchStoreService.isInactiveOrder(order)) throw new UserError('Cannot modify pickup for inactive orders');
+
+      const pickupEvent = await Repositories.merchOrderPickupEvent(txn).findByUuid(pickupEventUuid);
+      if (!pickupEvent) throw new NotFoundError('Order pickup event not found');
+      if (MerchStoreService.isLessThanTwoDaysBeforePickupEvent(pickupEvent)) {
+        throw new UserError('Cannot change order pickup to an event that starts in less than 2 days');
+      }
+      const orderInfo = await MerchStoreService.buildOrderUpdateInfo(order, pickupEvent, txn);
+      await this.emailService.sendOrderPickupUpdated(user.email, user.firstName, orderInfo);
+      return orderRepository.upsertMerchOrder(order, { pickupEvent });
+    });
+  }
+
+  private static isInactiveOrder(order: OrderModel): boolean {
+    return order.status === OrderStatus.FULFILLED || order.status === OrderStatus.CANCELLED;
+  }
+
+  /**
+   * Marks an order as missed. An order can be marked as missed only if it's previous status was PLACED,
+   * and its associated pickup event has already passed. Only admins can mark orders as missed
+   * @param uuid order uuid
+   * @returns updated order
+   */
+  public async markOrderAsMissed(uuid: Uuid, proxy: UserModel): Promise<OrderModel> {
```

Comment:
I think we talked a while back about not using `proxy` but maybe `admin` isn't the best alternative? It's better than `proxy` at least.
```suggestion
  public async markOrderAsMissed(uuid: Uuid, admin: UserModel): Promise<OrderModel> {
```

Reply:
I was hesitant to use `admin` since soon we'll have `STORE_MANAGER` and `STORE_DISTRIBUTOR` roles who aren't necessarily admins but can mark orders as missed. We can probably leave it as admin for now but change it later to something like `manager`? But yeah, something that conveys 'on behalf of' would be ideal.